src/js/views/patients/worklist/bulk-edit/bulk-edit_views.js

```diff
@@ -0,0 +1,436 @@
+      itemTemplateContext() {
+        return {
+          itemCount,
+        };
+      },
+      lists: [{ collection: menuOptions }],
+      align: 'right',
+      popWidth: 248,
+    });
+
+    optionlist.show();
+  },
+  onConfirmDelete() {
+    const modal = Radio.request('modal', 'show:small', {
+      bodyText: i18n.confirmDeleteFlowsModal.bodyText,
+      headingText: i18n.confirmDeleteFlowsModal.headingText,
+      submitText: i18n.confirmDeleteFlowsModal.submitText,
+      buttonClass: 'button--red',
+      onSubmit: () => {
+        modal.destroy();
+        this.triggerMethod('delete', this.collection);
+      },
+    });
+  },
+});
+
+const BulkEditActionsBodyView = View.extend({
+  className: 'bulk-edit__body',
+  template: BulkEditBodyTemplate,
+  regions: {
+    state: '[data-state-region]',
+    owner: '[data-owner-region]',
+    dueDay: '[data-due-day-region]',
+    dueTime: '[data-due-time-region]',
+    duration: '[data-duration-region]',
+  },
+  onRender() {
+    this.showState();
+    this.showOwner();
+    this.showDueDay();
+    this.showDueTime();
+    this.showDuration();
+  },
+  getBulkState() {
+    const stateId = this.collection.at(0).getState().id;
+    const hasDifferentState = this.collection.some(item => {
+      return item.getState().id !== stateId;
+    });
+
+    return hasDifferentState ? null : stateId;
+  },
+  getBulkOwner() {
+    const owner = this.collection.at(0).getOwner();
+    const hasDifferentOwner = this.collection.some(item => {
+      return item.getOwner().id !== owner.id;
+    });
+
+    return hasDifferentOwner ? null : owner;
+  },
+  getBulkGroups() {
+    const patients = new Backbone.Collection();
+    
+    this.collection.each(item => {
+      patients.add(item.getPatient());
+    });
+    
+    const groups = patients.at(0).getGroups();
+
+    patients.each(patient => {
+      const patientGroups = patient.getGroups();
+      const intersectedGroups = _.intersection(patientGroups.models, groups.models);
+
+      groups.reset(intersectedGroups);
+    });
+
+    return groups;
+  },
+  getBulkDueDate() {
+    const date = this.collection.at(0).get('due_date');
+    const hasDifferentDate = this.collection.some(item => {
+      return item.get('due_date') !== date;
+    });
+
+    return hasDifferentDate ? 'multiple' : date;
+  },
+  getBulkDueTime() {
+    const time = this.collection.at(0).get('due_time');
+    const hasDifferentTime = this.collection.some(item => {
+      return item.get('due_time') !== time;
+    });
+
+    return hasDifferentTime ? 'multiple' : time;
+  },
+  getBulkDuration() {
+    const duration = this.collection.at(0).get('duration');
+    const hasDifferentDuration = this.collection.some(item => {
+      return item.get('duration') !== duration;
+    });
+
+    return hasDifferentDuration ? 'multiple' : duration;
+  },
+  showState() {
```

Comment:
so what if in the app that creates this view it does all of the getting of data and sets the model for this view to include data like:
```js
{
 stateId: null,
 stateMulti: true,
 due_date: null,
 dueMulti: false,
}
```
then in this view we have:
```js
modelEvents: {
  'change:stateMulti': 'showState',
},
getStateComponent() {
  const stateId = this.model.get('stateId');
  if (this.model.get('stateMulti')) {
    return new StateComponent({ 
      stateId,
      viewOptions: {
        className: 'button-secondary w-100',
        template: MultipleStateTemplate,
      },
    });
  }
  return new StateComponent({ stateId });
},
showState() {
  const stateComponent = this.getStateComponent();
  
  this.listenTo(stateComponent, 'change:state', state => {
    this.model.set({ stateId: state, stateMulti: false });
  });

  this.showChildView('state', stateComponent);
}
```

This way we're only overriding `viewOptions` if there's multiple values, and once something is chosen and/or the data is the same, it just rebuilds the regular component and we don't have to duplicate as much of the component functionality.

Reply:
Good call. That would clean things up a lot. And you're right, we only need to know about the multiple attributes when the view first initializes.